### PR TITLE
Add option choose method to deliver emails in EmailNotifier

### PR DESCRIPTION
**Pull request by eitoball**
After updating Rails to 4.2.0, I get deprecation warnings like:

```
DEPRECATION WARNING: `#deliver` is deprecated and will be removed in Rails 5. Use `#deliver_now` to deliver immediately or `#deliver_later` to deliver through Active Job. 
```

So, I would like to add an option to choose method to deliver emails.

### CODE REVIEW

**Comment by mcfiredrill** [comment]
Maybe it could default to `:deliver_now`?

**Comment by smartinez87** [comment]
hi @eitoball, thanks for the contribution!

Could you please rebase it against master so I can merge it?

**Comment by eitoball** [comment]
@smartinez87 Rebased. Thank you for checking out!

File: test/exception_notifier/email_notifier_test.rb [hunk 1]
```diff
@@ -149,4 +149,27 @@ class EmailNotifierTest < ActiveSupport::TestCase
...
+      # used. If methods like "#deliver_now" is used, it should not
+      # give any warnings.
+      assert_silent do
+        email_notifier.call(@exception)
+      end
+    end
+  end
+
+  # shamelessly copied from Mail::Field in mikel/mail
+  def unfold(string)
```
Comment (smartinez87) [inline test/exception_notifier/email_notifier_test.rb]: where is this method used? I don't understand why it is needed...

File: lib/exception_notifier/email_notifier.rb [hunk 2]
```diff
@@ -172,7 +167,8 @@ def self.default_options
         :mailer_settings => nil,
         :email_headers => {},
         :mailer_parent => 'ActionMailer::Base',
-        :template_path => 'exception_notifier'
+        :template_path => 'exception_notifier',
+        :deliver_with => :deliver
```
Comment (smartinez87) [inline lib/exception_notifier/email_notifier.rb]: can you make it default to `deliver_now` instead?

Comment (eitoball) [inline lib/exception_notifier/email_notifier.rb]: Done!

File: test/exception_notifier/email_notifier_test.rb (hunk 1 above)
Comment (eitoball) [inline test/exception_notifier/email_notifier_test.rb]: The method was used to make test pass like d13c366d6a36d85a2800ae470e15e8eb3e7b7abb. When I rebased, I took this commit, but forgot delete this method.

**Comment by smartinez87** [comment]
@eitoball excellent, thanks!

**Pull request merged**